R/run_eggla_gwas.R

```diff
@@ -508,6 +518,42 @@ run_eggla_gwas <- function(
         "--out", results_file
       ), collapse = " "))
 
+      if (!quiet) message(sprintf("[%s] Computing PLINK2 frequencies ...", basename(vcf)))
+      system(paste(c(
+        bin_path[["plink2"]],
+        "--vcf", vcf_file, "dosage=DS",
+        "--mach-r2-filter",
+        "--threads", threads,
+        "--freq",
+        if (file.exists(sprintf("%s.samples", basename_file))) c("--keep", sprintf("%s.samples", basename_file)),
+        "--silent",
+        "--out", results_file
+      ), collapse = " "))
+
+      if (!quiet) message(sprintf("[%s] Computing PLINK2 missing rate ...", basename(vcf)))
+      system(paste(c(
+        bin_path[["plink2"]],
+        "--vcf", vcf_file, "dosage=DS",
+        "--mach-r2-filter",
+        "--threads", threads,
+        "--missing",
+        if (file.exists(sprintf("%s.samples", basename_file))) c("--keep", sprintf("%s.samples", basename_file)),
+        "--silent",
+        "--out", results_file
+      ), collapse = " "))
+
+      if (!quiet) message(sprintf("[%s] Computing PLINK2 Hardy-Weinberg test ...", basename(vcf)))
+      system(paste(c(
+        bin_path[["plink2"]],
+        "--vcf", vcf_file, "dosage=DS",
+        "--mach-r2-filter",
+        "--threads", threads,
+        "--hardy",
+        if (file.exists(sprintf("%s.samples", basename_file))) c("--keep", sprintf("%s.samples", basename_file)),
+        "--silent",
+        "--out", results_file
+      ), collapse = " "))
```

Comment:
Hardy-Weinberg equilibrium test report by PLINK2 using the same input as for GLM.

Reply:
@annihei It should use the dosage field of the VCF, see the command here.

Reply:
Yes, I tested the calculation with dosage and it is not intuitive to me how it is calculated as the dosage does not have detailed information on the number of homozygous or heterozygous individuals on a site. I didn't find dosage and HWE mentioned at the same time in the documentation.

Reply:
I have to admit, I did not spend time to check this particular aspect and if the `"dosage=DS"` identifier was really taken into account when computing Hardy-Weinberg equilibrium test.

How do you want to proceed for this?

Reply:
I agree with you and it sounds like everyone else is fine with it if we drop the HWE_P from the summary stats. I haven't been able to check the sample sizes N0,N1,N2 yet but they are probably not crucial either if they are difficult to define.